packages/sdk/src/server/index.ts

```diff
@@ -745,25 +835,97 @@ export class ServerClient {
    *       "Accept": "application/json",
    *     },
    *   },
+   *   "oauth",
    * );
    * console.log(response);
    * ```
    */
-  public async invokeWorkflow(url: string, opts: RequestOptions = {}): Promise<unknown> {
+  public async invokeWorkflow(urlOrEndpoint: string, opts: RequestOptions = {}, authType: HTTPAuthType = HTTPAuthType.None): Promise<unknown> {
     const {
       body,
       headers = {},
     } = opts;
 
+    const url = this.buildWorkflowUrl(urlOrEndpoint);
+
+    let authHeader: string | undefined;
+    switch (authType) {
+    // It's expected that users will pass their own Authorization header in the static bearer case
+    case HTTPAuthType.StaticBearer:
+      authHeader = headers["Authorization"];
+      break;
+    case HTTPAuthType.OAuth:
+      authHeader = await this.oauthAuthorizationHeader();
+      break;
+    default:
+      break;
+    }
+
     return this.makeRequest("", {
       ...opts,
       baseURL: url,
       method: opts.method || "POST", // Default to POST if not specified
+      headers: authHeader
+        ? {
+          ...headers,
+          "Authorization": authHeader,
+        }
+        : headers,
+      body,
+    });
+  }
+
+  /**
+   * Invokes a workflow for a Pipedream Connect user in a project
+   *
+   * @param url - The URL of the workflow's HTTP interface.
+   * @param externalUserId — Your end user ID, for whom you're invoking the workflow.
+   * @param opts - The options for the request.
+   * @param opts.body - The body of the request. It must be a JSON-serializable
+   * value (e.g. an object, null, a string, etc.).
+   * @param opts.headers - The headers to include in the request. Note that the
+   * Authorization header will always be set with an OAuth access token
+   * retrieved by the client.
+   *
+   * @returns A promise resolving to the response from the workflow.
+   *
+   * @example
+   *
+   * ```typescript
+   * const response = await client.invokeWorkflowForExternalUser(
+   *   "https://your-workflow-url.m.pipedream.net",
+   *   "your-external-user-id",
+   *   {
+   *     body: {
+   *       foo: 123,
+   *       bar: "abc",
+   *       baz: null,
+   *     },
+   *     headers: {
+   *       "Accept": "application/json",
+   *     },
+   *   },
+   * );
+   * console.log(response);
+   * ```
+   */
+  public async invokeWorkflowForExternalUser(url: string, externalUserId: string, opts: RequestOptions = {}): Promise<unknown> {
```

Comment:
Any particular reason why we're not validating the url here?

Reply:
adding some validation logic in general in `buildWorkflowUrl`, which should also address some of the other validation feedback above